django/db/models/base.py

```diff
@@ -1617,6 +1617,14 @@ def _check_swappable(cls):
                         id="models.E002",
                     )
                 )
+        if cls._meta.db_table_comment and not connection.features.supports_comments:
+            errors.append(
+                checks.Warning(
+                    f"{connection.display_name} does not support db_table_comment.",
+                    obj=cls,
+                    id="models.W045",
+                )
+            )
```

Comment:
This should be moved to the separate method and take `databases` and `required_db_features` into account.

Reply:
This warning is also untested.

Reply:
Tests added.